### PR TITLE
Remove EntityBase from sprinkler

### PR DESCRIPTION
**Pull request by jesserockz**
# What does this implement/fix?

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by jesserockz** [comment]
@kbx81 can you test this out? Should not require config changes

**Comment by jesserockz** [comment]
Ah didnt realise you set the default name to the switch name if only 1. Will fix that

**Comment by kbx81** [comment]
Yes, there's a little "gotcha" there -- let me know when you're good, I'll give it a quick test here. 😄 Thank you!

**Review by kbx81 — APPROVED** [review]
Ran it through my suite of test configs, all looks good. Thanks for your help!

File: esphome/components/sprinkler/__init__.py [hunk 1]
```diff
@@ -103,6 +103,10 @@ def validate_min_max(config):
 
 def validate_sprinkler(config):
     for sprinkler_controller_index, sprinkler_controller in enumerate(config):
+        if CONF_NAME not in sprinkler_controller:
+            id = sprinkler_controller[CONF_ID]
+            sprinkler_controller[CONF_NAME] = id.id
+
```
Comment (jesserockz) [inline esphome/components/sprinkler/__init__.py]: Don't actually need this
```suggestion
```

**Pull request merged**